src/Autofac.Extras.FakeItEasy/AutoFake.cs

```diff
@@ -79,7 +81,16 @@ public void Dispose()
         /// <typeparam name="T">The type of the service.</typeparam>
         /// <param name="parameters">Optional parameters.</param>
         /// <returns>The service.</returns>
-        public T Resolve<T>(params Parameter[] parameters) => this._currentScope.Resolve<T>(parameters);
+        public T Resolve<T>(params Parameter[] parameters)
+        {
+            T instance = default(T);
+            if (parameters.Length == 0)
+            {
+                instance = this.InstanceFromCurrentOrParentScopes<T>();
```

Comment:
Might people be surprised that the parameters (whatever those might be) aren't considered when we're trying to get an instance from a parent scope?

Reply:
I assumed that parameters are used to creating new instances rather than obtaining registered ones, but you're right because it is not clear at all and people might use it in different ways. I am also not used to autofac, so i will need to spend some time understanding it a little better and to find a solution where we can retrieve instances by filtering parameters too.

Reply:
Interesting. As you say, we don't really know enough.

I'd assumed you were using Autofac.Extras.FakeItEasy because you already were using autofac. Personal styles differ, as do projects, but me, I prefer not to introduce additional fake-management libraries into my tests. Instead I just... make up Fakes and use them when I need to. Maybe I just have less complex tests and can get away with it, or I have a higher tolerance for some kinds of repetition. 

I don't think I'm going to be a resource for figuring out how the autofac stuff does or should work, but would be happy to talk about your testing pain in general, if you think it might help.

Reply:
Not going to lie, but I mostly also prefer to not use an IoC container to deal my tests, in fact I code to make them simple and clean.

Unfortunately I dealt some legacy codes that refactoring was off the table, but I would like to keep at least the habit to write tests about my changes. I know that the burden of storing fakes/mocks values is not related to FakeItEasy neither any other similar lib, so at least I've been trying to seek for a solution to deal with fake instances when its quantity becomes troublesome. So it brought me here.

Reply:
Unfortunately, this is also not something the Autofac maintainers will be able to help with. Like some other integration libraries we've had for a while, this is something that came from the community but isn't getting maintained by the community. I don't think any of the remaining Autofac owners are really FakeItEasy users, and if we are... I don't think any of use use IoC containers to manage mocks in tests, either. (I use Moq, but I also don't use the Autofac Moq integration.)

When I have challenging mock setup, I usually create a little static factory method like this:

```c#
private static void CreateMyThing()
{
  var firstMock = Mock.Of<IFirst>();
  var secondMock = Mock.Of<ISecond>();
  return new MyThing(firstMock, secondMock);
}
```

If I need the mocks back later to assert, I get them from the object because most of the time there's a property with the dependency.

```c#
var thing = CreateMyThing();
var firstMock = Mock.Get(thing.First);
firstMock.Verify(...);
```

Barring that, I'll maybe have the factory method take in optional parameters and set up defaults for the rest.

```c#
private static void CreateMyThing(IFirst first = null, ISecond second = null)
{
  var firstMock = first ?? Mock.Of<IFirst>();
  var secondMock = second ?? Mock.Of<ISecond>();
  return new MyThing(firstMock, secondMock);
}
```

Stuff like that. I don't think I have any unit test in any project where I try to auto-fill a bunch of mocks. If it's too complicated to set up in a little factory method, it's refactor time. (Yes, I see that's not an option for you, but maybe the factory method still is.)

Anyway, again, sorry there won't really be much help coming from the Autofac side.

Technically we (Autofac folks) should probably put this library in maintenance mode until someone from the community steps up to own it outright. It's not really fair to make it seem like we're actively maintaining it when we're not.

Reply:
You have a good point. I didn't realize that factories could be helpful in my scenario. I think it will work properly like some instance builders I've write. Thank you.

About the repository, I think it is a properly decision. Unfortunately, it could be harmful to integrate pull requests with our lack of knowledge about both libs together.

If you think this current PR could be helpful someway or someone in a near future, I can keep it open here; otherwise we can just close it. Thanks you guys again!

Reply:
Sounds good. I think we'll close this for now. Good luck!